yt_dlp/extractor/generic.py

```diff
@@ -3410,6 +3411,12 @@ def _real_extract(self, url):
             return self.playlist_from_matches(
                 rutube_urls, video_id, video_title, ie=RutubeIE.ie_key())
 
+        # Look for Glomex embeds
+        glomex_urls = list(GlomexEmbedIE._extract_urls(webpage, url))
+        if glomex_urls:
+            return self.playlist_from_matches(
+                glomex_urls, video_id, video_title, ie=GlomexEmbedIE.ie_key())
```

Comment:
Can you add tests? Preferably one for each type of embed we try to detect

Reply:
I had not realized there could be tests for `_extract_urls`. I will look into it.

Reply:
I added a test case for the first integration method (iframe). Unfortunately currently I can not find sites using the other two methods.
It can be run with `test_download.py TestDownload.test_Generic_122`.

It seems the regex runs into catastrophic backtracking, while this did not happen before the last changes. I will (also) look into it.

Reply:
It works after reverting part of the last change:

```patch
--- a/yt_dlp/extractor/glomex.py
+++ b/yt_dlp/extractor/glomex.py
@@ -189,7 +189,7 @@ class GlomexEmbedIE(GlomexBaseIE):
                     (?P<_q5>%(quot_re)s)(?P<integration_js>(?:(?!(?P=_q5)).)+)(?P=_q5)\s*(?(_stjs1);|,)?|
                 (?P<_stjs2>dataset\.)?playlistId\s*(?(_stjs2)=|:)\s*
                     (?P<_q6>%(quot_re)s)(?P<id_js>(?:(?!(?P=_q6)).)+)(?P=_q6)\s*(?(_stjs2);|,)?|
-                [^>]*?
+                (?:\s|.)*?
             )+</script>
         )''' % {'quot_re': r'["\']', 'url_re': rf'(?:https?:)?{re.escape(cls._BASE_PLAYER_URL)}\?(?:(?!(?P=_q1)).)+'}
 
```

IIRC, matching anything except `>` is not safe for consuming the `<script>` content. And it probably backlashes with the same approach used in the previous alternative pattern for consuming attributes up to the end of the HTML tag.

Shall I push this partial revert?

Reply:
yes, go ahead